### PR TITLE
Initial changes to move to Python 3.6

### PR DESCRIPTION
**Pull request by antgonza**
Tests on this are going to fail but it's going to it's own branch so we can minimize reviews complexity. 

This PR:
- installs qiita and its dependencies in python 3.6 (which is the version that conda and qiime2 are moving to)
- makes sure that qiita can be ran and the webserver can be started
- makes sure that qiita-env clean-tests works
- flake8 passes

### CODE REVIEW

File: qiita_ware/ebi.py [hunk 1]
```diff
@@ -704,7 +704,7 @@ def write_xml_file(self, element, fp):
         fp : str
             The filepath to which the XML will be written
         """
-        create_dir(self.xml_dir)
+        makedirs(self.xml_dir)
         ET.ElementTree(element).write(fp, encoding='UTF-8')
```
Comment (charles-cowart) [inline qiita_ware/ebi.py]: This wouldn't be UTF-8 by default?

File: qiita_db/__init__.py [hunk 2]
```diff
@@ -6,26 +6,26 @@
...
-import logger
-import meta_util
-import ontology
-import portal
-import reference
-import software
-import study
-import user
-import processing_job
+from . import base
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_db/__init__.py]: IMOO we don't need to make any comments

File: qiita_db/base.py [hunk 3]
```diff
@@ -175,17 +175,15 @@ def __init__(self, id_):
         # the User object) are strings. Moreover, some integer IDs are passed
         # as strings (e.g., '5'). Therefore, explicit type-checking is needed
         # here to accommodate these possibilities.
-        if not isinstance(id_, (int, long, str, unicode)):
+        if not isinstance(id_, (int, str)):
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_db/base.py]: ... same opinion ... no change/comment.

File: qiita_db/handlers/oauth2.py [hunk 4]
```diff
@@ -90,13 +90,14 @@ def wrapper(handler, *args, **kwargs):
...
-                                               db_token['user'])
+        if db_token[b'grant_type'] == b'password':
+            limit_key = '%s_%s_daily_limit' % (
+                db_token[b'client_id'].decode('ascii'),
+                db_token[b'user'].decode('ascii'))
             limiter = r_client.get(limit_key)
             if limiter is None:
                 # Set limit to 5,000 requests per day
-                r_client.setex(limit_key, 5000, 86400)
+                r_client.setex(limit_key, 86400, 5000)
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_db/handlers/oauth2.py]: Yes, for some reason the new version of redis works like this: time, value ...

File: qiita_db/handlers/oauth2.py [hunk 5]
```diff
@@ -212,19 +213,23 @@ def set_token(self, client_id, grant_type, user=None, timeout=3600):
...
+        r_client.hmset(token, token_info)
+        r_client.expire(token, timeout)
         if grant_type == 'password':
             # Check if client has access limit key, and if not, create it
             limit_key = '%s_%s_daily_limit' % (client_id, user)
             limiter = r_client.get(limit_key)
             if limiter is None:
                 # Set limit to 5,000 requests per day
-                r_client.setex(limit_key, 5000, 86400)
+                r_client.setex(limit_key, 86400, 5000)
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_db/handlers/oauth2.py]: yup

File: qiita_db/handlers/tests/test_oauth2.py [hunk 6]
```diff
@@ -17,20 +17,29 @@ class OAuth2BaseHandlerTests(TestHandlerBase):
...
+            'grant_type': 'password',
+            'user': 'test@foo.bar'
+        }
+        r_client.hmset(self.user_token, token_info)
         r_client.expire(self.user_token, 5)
+
         # Create test access limit token
         self.user_rate_key = 'testuser_test@foo.bar_daily_limit'
-        r_client.setex(self.user_rate_key, 2, 5)
+        r_client.setex(self.user_rate_key, 5, 2)
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_db/handlers/tests/test_oauth2.py]: ... same ...

File: qiita_db/handlers/tests/test_oauth2.py [hunk 7]
```diff
@@ -45,7 +54,7 @@ def test_authenticate_header_username(self):
 
         # Check rate limiting works
         self.assertEqual(int(r_client.get(self.user_rate_key)), 1)
-        r_client.setex('testuser_test@foo.bar_daily_limit', 0, 2)
+        r_client.setex('testuser_test@foo.bar_daily_limit', 1, 0)
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_db/handlers/tests/test_oauth2.py]: Yes, before it was giving it 2 seconds, now just 1 ...

File: qiita_ware/ebi.py (hunk 1 above)
Comment (antgonza) [reply qiita_ware/ebi.py]: Not sure ... but I rather don't change anything I'm not sure about yet (this line didn't change )

**Pull request merged**